### PR TITLE
T255429 New demo site 

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Link : https://phabricator.wikimedia.org/T255429

**Design** has been confirmed by the designer

**Dev** please review the CI and webpack code changes, and don't be too picky on the demo code as it is modified from a static demo html

To preview this demo site, visit https://hueitan.github.io/sudhanshugautam3.github.io/

| Mobile View (1) | Mobile View (2) |
| --- | --- |
| Screen Shot 2020-07-22 at 15 15 31 | Screen Shot 2020-07-22 at 15 15 38 |

| Desktop View (1) | Desktop View (2) |
| --- | --- |
| Screen Shot 2020-07-22 at 15 15 51 | Screen Shot 2020-07-22 at 15 15 49 |

### CODE REVIEW

**Comment by medied** [comment]
After the re-write the demo is now rendering much better across my devices (Chrome, Firefox in desktop and iphone), working lovely on mobile, good work Huei. 

Some observations:

* From the Phabricator ticket description I see this request: "Shows how Preview links are distinguished from other hyperlinks". Is this a good moment to add them?
* Do we need to update README images? I guess that's not really necessary, they can serve as the memory of the first demo site
* `npm run build` produces assets as expected
* I can run locally on desktop and mobile as expected

File: demo/articles/english.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                </p>
+                <p>
+                    Realising the serious threat to the wildlife, the colonists – French Administration – in 1935 and later the Government of the Republic of CRA, enacted laws and created National parks and preserves, which covered 16.6% of the country. The three most coveted national parks are the Manovo-Gounda St. Floris National Park with its reported "greatest concentrations of hippos in the world", the<span class="wmf-wp-with-preview" data-wp-title="Bamingui-Bangoran_National_Park_and_Biosphere_Reserve" data-wikipedia-preview> Bamingui-Bangoran National</span> Park in the north; and the Dzanga-Sangha Reserve which covers rain forests. The Manovo-Gounda-Saint-Floris National Park, in particular was inscribed to the UNESCO List of World Heritage Sites in 1988 in recognition of the diversity of life present within it in respect of its wealth of flora and fauna.
```
Comment (medied) [inline demo/articles/english.html:40]: Inserting a space at "the<span..."

```suggestion
                    Realising the serious threat to the wildlife, the colonists – French Administration – in 1935 and later the Government of the Republic of CRA, enacted laws and created National parks and preserves, which covered 16.6% of the country. The three most coveted national parks are the Manovo-Gounda St. Floris National Park with its reported "greatest concentrations of hippos in the world", the <span class="wmf-wp-with-preview" data-wp-title="Bamingui-Bangoran_National_Park_and_Biosphere_Reserve" data-wikipedia-preview> Bamingui-Bangoran National</span> Park in the north; and the Dzanga-Sangha Reserve which covers rain forests. The Manovo-Gounda-Saint-Floris National Park, in particular was inscribed to the UNESCO List of World Heritage Sites in 1988 in recognition of the diversity of life present within it in respect of its wealth of flora and fauna.
```

File: demo/articles/spanish.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                        </div>
+                    </div>
+                </a>
+            </div>
+            <div class="title">
+                <p>Nikola Tesla</p>
+            </div>
+            <div class="content">
+                <p>
+                    Nikola Tesla fue un inventor, ingeniero eléctrico, ingeniero mecánico y físico estadounidense de origen serbocroata.1​2​ Se le conoce sobre todo por sus numerosas invenciones en el campo del electromagnetismo, desarrolladas a finales del siglo XIX y principios del siglo XX. Las patentes de Tesla y su trabajo teórico ayudaron a forjar las bases de los sistemas modernos para el uso de la energía eléctrica por corriente alterna (CA), incluyendo el sistema polifásico de distribución eléctrica yel <span class="wmf-wp-with-preview" data-wikipedia-preview>motor de corriente alterna</span>, que contribuyeron al surgimiento de la Segunda Revolución Industrial.
```
Comment (medied) [inline demo/articles/spanish.html:49]: Some minor typos:
* serbocroata.1​2 Se​ ---> serbocroata. Se
* yel <span ---> y el <span

```suggestion
                    Nikola Tesla fue un inventor, ingeniero eléctrico, ingeniero mecánico y físico estadounidense de origen serbocroata.​ Se le conoce sobre todo por sus numerosas invenciones en el campo del electromagnetismo, desarrolladas a finales del siglo XIX y principios del siglo XX. Las patentes de Tesla y su trabajo teórico ayudaron a forjar las bases de los sistemas modernos para el uso de la energía eléctrica por corriente alterna (CA), incluyendo el sistema polifásico de distribución eléctrica y el <span class="wmf-wp-with-preview" data-wikipedia-preview>motor de corriente alterna</span>, que contribuyeron al surgimiento de la Segunda Revolución Industrial.
```

File: demo/articles/spanish.html [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                </p>
+                <p>
+                    Su carácter, su enfrentamiento con <span class="wmf-wp-with-preview" data-wp-title="Thomas_Alva_Edison" data-wikipedia-preview>Edison</span> y el halo de misterio que rodea a algunos de sus descubrimientos, hicieron que Tesla se convirtiera en un científico muy popular a partir de la década de 1990, con una abundante bibliografía disponible acerca de su vida y de su obra.
+                </p>
+                <p>
+                    Nikola Tesla, de etnia serbia,5&#8203;6&#8203; nació en el pueblo de <span class="wmf-wp-with-preview" data-wikipedia-preview>Smiljan</span> (actualmente en Croacia), en el entonces Imperio austrohúngaro, y tiempo después se nacionalizaría estadounidense
```
Comment (medied) [inline demo/articles/spanish.html:55]: * serbia,5&#8203;6&#8203; nació  ---> serbia, nació 
```suggestion
                    Nikola Tesla, de etnia serbia, nació en el pueblo de <span class="wmf-wp-with-preview" data-wikipedia-preview>Smiljan</span> (actualmente en Croacia), en el entonces Imperio austrohúngaro, y tiempo después se nacionalizaría estadounidense
```

**Comment by hueitan** [comment]
> * From the Phabricator ticket description I see this request: "Shows how Preview links are distinguished from other hyperlinks". Is this a good moment to add them?

As discussed with design, it currently shows different than the normal hyperlinks now, so we are safe here.

> * Do we need to update README images? I guess that's not really necessary, they can serve as the memory of the first demo site

It's not necessary here, we need to update the screenshot once the gallery is done for sure.

**Comment by medied** [comment]
> ...we need to update the screenshot once the gallery is done for sure.

You're right. I suggest we update README with gallery screenshots once the licensing info is ready (so with [this ticket](https://phabricator.wikimedia.org/T240007))

**Pull request merged**